contentcuration/contentcuration/frontend/channelEdit/vuex/contentNode/index.js

```diff
@@ -29,13 +29,26 @@ export default {
        *
```

Comment:
Sorry, did not do a correct and full update of the code comments here, will add.

In short, I eliminated the nested mapping loading from the backend, and we are now just representing the relationships with `{target_node: <target_node>, prerequisite: <prerequisite>}` objects in the data returned from the backend.

The lookups being done are always by id, so a mappable object seemed like a better fit for the getters that we were using (as opposed to doing find operations on a list).

In order to make the lookups reversable, there is a parallel `previousSteps` map. This could certainly be a getter that inverts the nextStepsMap. It was probably a premature optimization to denormalize it.

Reply:
Okay I assumed that it might be somehow related to data returned from the backend. Thanks for clarification.

Reply:
Also now when I get that there won't be more nested levels, my main concerns from this comment are resolved.